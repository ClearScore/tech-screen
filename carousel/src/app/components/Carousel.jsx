import React, { Component } from 'react';
import styled from 'styled-components';
import * as d3 from 'd3';
import ScoreIndicator from './ScoreIndicator/ScoreIndicator';
import DebtIndicator from './DebtIndicator/DebtIndicator';

class Carousel extends Component {
  state = {
    index: 0,
    size: 500
  };

  componentDidMount() {
    window.setInterval(this.changeSlide, 3000);
    window.setTimeout(this.drawArc, 500);
    // this.drawArc();
  }

  changeSlide = () => {
    let index = this.state.index === 0 ? 1 : 0;
    this.setState({ index });
  };

  drawArc = () => {
    const context = this.setContext();
    this.setBackground(context);
    this.setForeground(context);
  };

  renderSlides = () => {
    if (this.state.index === 0)
      return <ScoreIndicator score={this.props.score} />;
    else return <DebtIndicator debt={this.props.debt} />;
  };

  setContext() {
    const size = this.state.size;
    return d3
      .select(this.refs.arc)
      .append('svg')
      .attr('height', size)
      .attr('width', size)
      .attr('id', 'd3-arc')
      .append('g')
      .attr('transform', `translate(250, 250)`);
  }
  setBackground(context) {
    return context
      .append('path')
      .datum({ endAngle: this.tau })
      .style('fill', 'transparent')
      .attr('d', this.arc());
  }

  tau = Math.PI * 2;

  arc() {
    const size = this.state.size;
    return d3
      .arc()
      .innerRadius(size / 2 - 5)
      .outerRadius(size / 2)
      .startAngle(0);
  }
  setForeground(context) {
    const score = this.props.score;
    return context
      .append('path')
      .datum({ endAngle: this.tau * score / 700 })
      .style('fill', 'blue')
      .attr('d', this.arc());
  }
  render() {
    return (
      <div className="carousel">
        <div ref="arc" className="arc" />
        {this.renderSlides()}
      </div>
    );
  }
}

export default Carousel;
