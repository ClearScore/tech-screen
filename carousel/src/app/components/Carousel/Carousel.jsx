import React, { Component } from 'react';
import styled from 'styled-components';
import root from 'window-or-global';
import ScoreIndicator from './../ScoreIndicator/ScoreIndicator';
import DebtIndicator from './../DebtIndicator/DebtIndicator';
import Arc from './../Arc/Arc';

class Carousel extends Component {
  state = {
    index: 0
  };
  componentDidMount() {
    window.setInterval(this.changeSlide, 3000);
  }
  changeSlide = () => {
    let index = this.state.index === 0 ? 1 : 0;
    this.setState({ index });
  };
  renderSlides = () => {
    if (this.state.index === 0)
      return <ScoreIndicator score={this.props.score} />;
    else return <DebtIndicator debt={this.props.debt} />;
  };
  render() {
    // const size = 500;
    const score = this.props.score;
    return (
      <div className="carousel">
        {/* <Arc size={size} score={score} /> */}
        {this.renderSlides()}
      </div>
    );
  }
}

export default Carousel;
