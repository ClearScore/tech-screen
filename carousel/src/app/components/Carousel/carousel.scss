@import '../../../styles/utils/colours';
@import '../../../styles/utils/breakpoints';

$carousel-size-desktop: 600px;
$carousel-size-mobile: 300px;

.carousel {
  margin: 0 auto;
  width: $carousel-size-desktop;
  height: $carousel-size-desktop;
  // Cheat so we don't see a slide-in glitch when the carousel first renders
  animation: fadeIn 3s;

  @include breakpoint(small) {
    width: $carousel-size-mobile;
    height: $carousel-size-mobile;
  }

  &__slide-container {
    position: absolute;
    display: inline-block;
    border: 1px solid $secondary;
    width: $carousel-size-desktop;
    height: $carousel-size-desktop;
    border-radius: 50%;
    overflow: hidden;
    cursor: pointer;
    transform: translateX(-1000px) scale(0.5);
    opacity: 0;
    animation: slideOut 2s;

    @include breakpoint(small) {
      width: $carousel-size-mobile;
      height: $carousel-size-mobile;
      animation: slideOutMobile 2s;
    }

    &:before {
      content: '';
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      margin: -30px;
      filter: blur(18px);
      z-index: -1;
    }

    &.visible {
      transform: translateX(0) scale(1);
      opacity: 1;
      display: inline-block;
      animation: slideIn 2s;
      @include breakpoint(small) {
        animation: slideInMobile 2s;
      }
    }
  }

  &__slide {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    color: $secondary;
    background-color: rgba(0,0,0,0.07);
    @include breakpoint(small) {
      background-color: rgba(0,0,0,0.15);
    }
  }

  &__indicators {
    position: absolute;
    bottom: 40px;
    margin: 0;
    width: 100%;
    padding: 0;
    text-align: center;
    list-style: none;
    li {
      display: inline-block;
      margin-right: 5px;
      border: 1px solid $secondary;
      border-radius: 50%;
      width: 10px;
      height: 10px;

      &.active {
        background-color: $secondary;
      }
    }
  }

}


@keyframes slideIn {
	from {
    transform: translateX(500px) scale(0.5);
    opacity: 0;
  }
	to {
    transform: translateX(0) scale(1);
    opacity: 1;
  }
}

@keyframes slideOut {
  from {
    transform: translateX(0) scale(1);
    opacity: 1;
  }
  to {
    transform: translateX(-500px) scale(0.5);
    opacity: 0;
  }
}

@keyframes slideInMobile {
	from {
    transform: translateX(300px) scale(0.5);
    opacity: 0;
  }
	to {
    transform: translateX(0) scale(1);
    opacity: 1;
  }
}

@keyframes slideOutMobile {
  from {
    transform: translateX(0) scale(1);
    opacity: 1;
  }
  to {
    transform: translateX(-300px) scale(0.5);
    opacity: 0;
  }
}

@keyframes fadeIn {
  0% {
    opacity: 0;
    transform: scale(0.5);
  }
  50% {
    opacity: 0;
    transform: scale(0.5);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}
