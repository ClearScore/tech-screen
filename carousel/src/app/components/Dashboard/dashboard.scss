*,
*:before,
*:after {
  box-sizing: border-box;
}
body {
  background: no-repeat top;
  background-size: cover;
  background-image: url('/background--dashboard.png');
}

// background image position needs to be fixed
