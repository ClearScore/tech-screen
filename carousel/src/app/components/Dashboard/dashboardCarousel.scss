.carousel--container {
  width: 100%;
  margin: 0 auto;

  @media only screen and (min-width: 768px) {
    max-width: calc(100% / 3);
  }

  //TODO - these are hacks for quick test styling
  .slider-control-centerright,
  .slider-control-centerleft {
    display: none;
  }

  .slider-control-bottomcenter button {
    color: white !important; //eww important
  }
}

.carousel--slide {
  border-radius: 50%;
  height: auto;
  padding-top: 100%;
  border: 1px solid white;
  max-width: 100%;
  box-sizing: border-box;
}

.carousel--slide-text {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 100%;
  color: white;
  display: flex;
  align-items: center;
  justify-content: center;
}

