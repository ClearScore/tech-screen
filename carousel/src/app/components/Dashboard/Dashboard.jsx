import React, { Component } from 'react';
import styled from 'styled-components';
import Carousel from './../Carousel/Carousel';
import root from 'window-or-global';
import creditReport from '../../creditReport';
import './dashboard.scss';

const details = creditReport.creditReport.creditReportInfo;

class Dashboard extends Component {
  render() {
    return (
      <div className="dashboard">
        <Carousel score={details.score} debt={details.currentLongTermDebt} />
      </div>
    );
  }
}
export default Dashboard;
