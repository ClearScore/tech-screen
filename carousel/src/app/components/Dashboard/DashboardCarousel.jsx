import React from "react";
import "./dashboardCarousel.scss";
import Carousel from "nuka-carousel";
import bemHelper from "../../utils/bem";
import PropTypes from "prop-types";
import Dashboard from "./Dashboard";

const cn = bemHelper({ block: "carousel" });
class DashboardCarousel extends React.Component {
  state = {};

  render() {
    const { creditReport } = this.props;
    return (
      creditReport && (
        <Carousel className={cn(null, "container")}>
          <div className={cn(null, "slide")}>
            <div className={cn(null, "slide-text")}>
              <h1 className="padding-0 margin-0">
                Credit Score: {creditReport.score}
              </h1>
            </div>
          </div>
          <div className={cn(null, "slide")}>
            <div className={cn(null, "slide-text")}>
              <h1 className="padding-0 margin-0">
                Long term debt: ${creditReport.currentLongTermDebt}
              </h1>
            </div>
          </div>
        </Carousel>
      )
    );
  }
}

DashboardCarousel.propTypes = {
  creditReport: PropTypes.object.isRequired
};

export default DashboardCarousel;
