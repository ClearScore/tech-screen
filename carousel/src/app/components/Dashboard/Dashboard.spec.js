import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from './Dashboard';
import Carousel from './../Carousel/Carousel';
import toJson from 'enzyme-to-json';

describe('Dashboard', () => {
  const wrapper = shallow(<Dashboard />);
  it('should correctly', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  it('should have Carousel component', () => {
    expect(wrapper.containsMatchingElement(<Carousel />)).toBe(true);
  });
});
