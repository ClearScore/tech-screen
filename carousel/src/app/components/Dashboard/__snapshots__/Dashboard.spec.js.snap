// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should correctly 1`] = `
<div
  className="dashboard"
>
  <Carousel
    debt={24682}
    score={514}
  />
</div>
`;
